contract/views/res_partner_view.xml

```diff
@@ -9,12 +9,12 @@
         <field type="xml" name="arch">
         <xpath expr="//div[@name='button_box']" position="inside">
             <button name="act_show_contract" type="object" class="oe_stat_button"
-                    icon="fa-book" context="{'contract_type': 'sale'}"
+                    icon="fa-book" context="{'default_contract_type': 'sale', 'contract_type': 'sale'}"
```

Comment:
You can add here (and in the next button) `attrs` for not showing this button if the partner is not a customer.

Reply:
done